src/Executor/Executor.php

```diff
@@ -222,12 +223,15 @@ private static function getOperationRootType(Schema $schema, OperationDefinition
     private static function executeFieldsSerially(ExecutionContext $exeContext, ObjectType $parentType, $sourceValue, $path, $fields)
     {
         $results = [];
+        $sourceValue = self::completePromiseIfNeeded($sourceValue);
+
         foreach ($fields as $responseName => $fieldASTs) {
             $fieldPath = $path;
             $fieldPath[] = $responseName;
             $result = self::resolveField($exeContext, $parentType, $sourceValue, $fieldASTs, $fieldPath);
 
             if ($result !== self::$UNDEFINED) {
+                $result = self::completePromiseIfNeeded($result);
```

Comment:
@mcg-web I played a bit with [your example](https://github.com/mcg-web/sandbox-dataloader-graphql-php). It appears that it doesn't actually defer fields execution. You will see what I mean if you change your query to the simplest possible variant:

```graphql
{ 
  character1: character(id: "1000") {
    name 
  } 
  character2: character(id: "1002") {
    name
  }
}
```
Solution with deferred fields would do 1 round-trip to storage, but both examples do 2. Query with `friends` actually never defers too - I put a breakpoint to `friends` resolver and `$character` argument never arrives as promise - only as array value.

The number of requests in original examples vary that much due to `cacheMap` in dataloader and lack of IN(?)-style request of friends in `without-dataloder` example.

I expected this because implementation of deferred fields just can't be that simple %) Though maybe I am missing something.

Also real test for deferreds should include at least following examples:

```graphql
  stories {
    author {
      name
    }
    likers {
      edges {
        node {
          name
        }
      }
    }
  }
```

It covers different cases, including one where deferred entities are on different levels of query.